tls/s2n_connection.c

```diff
@@ -477,6 +478,9 @@ int s2n_connection_release_buffers(struct s2n_connection *conn)
     POSIX_ENSURE(s2n_stuffer_is_consumed(&conn->in), S2N_ERR_STUFFER_HAS_UNPROCESSED_DATA);
     POSIX_GUARD(s2n_stuffer_resize(&conn->in, 0));
 
+    POSIX_ENSURE(s2n_stuffer_is_consumed(&conn->post_handshake.in), S2N_ERR_STUFFER_HAS_UNPROCESSED_DATA);
```

Comment:
I'm not sure what you mean by "handshake failure". This is an API that applications can call to free memory, not a part of the handshake. The documentation does call out it can return "S2N_FAILURE on failure", and we already set S2N_ERR_STUFFER_HAS_UNPROCESSED_DATA for the other buffers.